packages/block-library/src/tag-cloud/edit.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * External dependencies
+ */
+import { map, filter } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import {
+	PanelBody,
+	Placeholder,
+	ToggleControl,
+	SelectControl,
+	ServerSideRender,
+} from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { InspectorControls } from '@wordpress/editor';
+
+class TagCloudEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.setTaxonomy = this.setTaxonomy.bind( this );
+		this.toggleShowTagCounts = this.toggleShowTagCounts.bind( this );
+	}
+
+	getTaxonomies() {
+		const taxonomies = filter( this.props.taxonomies, 'show_cloud' );
+
+		return map( taxonomies, ( taxonomy ) => {
+			return {
+				value: taxonomy.slug,
+				label: taxonomy.name,
+			};
+		} );
+	}
+
+	setTaxonomy( taxonomy ) {
+		const { setAttributes } = this.props;
+
+		setAttributes( { taxonomy } );
+	}
+
+	toggleShowTagCounts() {
+		const { attributes, setAttributes } = this.props;
+		const { showTagCounts } = attributes;
+
+		setAttributes( { showTagCounts: ! showTagCounts } );
+	}
+
+	render() {
+		const { attributes } = this.props;
+		const { taxonomy, showTagCounts } = attributes;
+		const taxonomies = this.getTaxonomies();
+		const options = [
+			{
+				label: __( '- Select -' ),
+				value: '',
+			},
+			...taxonomies,
+		];
```

Comment:
nit: We could perform this array concatenation in `getTaxonomies()`. This would shorten `render()` which should help with readability.

Reply:
It can also be renamed it to `getTaxonomyOptions` to take into account the proposed change.